DigitalLearningSolutions.Data/Services/RegistrationService.cs

```diff
@@ -46,30 +51,51 @@ IConfiguration config
             this.emailService = emailService;
             this.centresDataService = centresDataService;
             this.config = config;
+            this.supervisorDelegateService = supervisorDelegateService;
+            this.frameworkNotificationService = frameworkNotificationService;
         }
 
         public (string candidateNumber, bool approved) RegisterDelegate(
             DelegateRegistrationModel delegateRegistrationModel,
             string userIp,
-            bool refactoredTrackingSystemEnabled
+            bool refactoredTrackingSystemEnabled,
+            int? supervisorDelegateId = null
         )
         {
-            var centreIpPrefixes =
-                centresDataService.GetCentreIpPrefixes(delegateRegistrationModel.Centre);
-            delegateRegistrationModel.Approved =
-                centreIpPrefixes.Any(ip => userIp.StartsWith(ip.Trim())) || userIp == "::1";
-
-            var candidateNumber =
-                registrationDataService.RegisterDelegate(delegateRegistrationModel);
+            var supervisorDelegateRecordIds = supervisorDelegateService.GetPendingSupervisorDelegateRecordsByEmail(

```

Comment:
`supervisorDelegateRecordIds` is a little vague... could we add `MatchingDelegate` or something on the end of the name?

Reply:
We could also extract this block into a method `GetPendingSupervisorDelegateIdsMatchingDelegate(delegateRegistrationModel)` to share with `RegistyerDelegateByCentre()`.